File: java/Palindrome.java
class Solution
{
	public boolean Palindrome(String str)
	{
		int left = 0;
		int right = str.length() - 1;

		while(left < right)
		{
			while(left < right && !Character.isLetterOrDigit(str.charAt(left))) left++;
			while(left < right && !Character.isLetterOrDigit(str.charAt(right))) right--;
			if(Character.toLowerCase(str.charAt(left)) != Character.toLowerCase(str.charAt(right))) return false;
			if(left < right) left++;
			if(left < right) right--;
		}

		return true;
	}
}

public class Palindrome
{
	public static void main(String[] args)
	{
		String str = "a #a c -bbc, a:  a ";
		Solution solution = new Solution();
		System.out.println(solution.Palindrome(str));
	}
}

File: c#/Palindrome.cs
using System;

public class Solution
{
	public bool Palindrome(string str)
	{
		int left = 0;
		int right = str.Length - 1;

		while(left < right)
		{
			while(left < right && !Char.IsLetterOrDigit(str[left])) left++;
			while(left < right && !Char.IsLetterOrDigit(str[right])) right--;
			if(Char.ToLower(str[left]) != Char.ToLower(str[right])) return false;
			if(left < right) left++;
			if(left < right) right--;
		}

		return true;
	}
}

public class LeetCode
{
	public static void Main()
	{
		string str = "a #a c -bbc, a:  a ";
		Solution solution = new Solution();
		Console.WriteLine(solution.Palindrome(str));
	}
}